Maintenance

Jyrno42 opened the pull request:
- Use Yarn
- use node 8-11 in CI
- Update dependencies
- Add prettier
- Fix linting issues
- Fix examples
- Fix webpack/babel config

simonschmidt commented:
:+1:

simonschmidt commented on CHANGELOG.md:
```diff
@@ -2,34 +2,41 @@
+- Update dependencies
+- Use yarn
+- Use node 8-12 in CI
+- Add prettier
```
:+1:

simonschmidt commented on examples/components/examples/Basic.js:
```diff
@@ -27,23 +26,28 @@ class BasicModalExample extends Component {
-        this.setState({
-            isOpen: !this.state.isOpen
-        });
+        this.setState((prevState) => ({
```
:+1:

simonschmidt commented:
Getting:

```
The resource from “http://127.0.0.1:8081/main.css” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff).
```

In browser console when checking out the examples in test server, not sure if new

Jyrno42 commented:
> Getting:
> 
> ```
> The resource from “http://127.0.0.1:8081/main.css” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff).
> Warning: Expected server HTML to contain a matching <div> in <div>.
> ```
> 
> In browser console when checking out the examples in test server, not sure if new

Thats actually OK, it is for the case where docs have been deployed to github pages where the css file actually exists.

simonschmidt commented:
Also in the example server, when closing a modal the page scrolls to top

Jyrno42 commented:
Can you try again now? I replaced the `position:fixed` and `overflow:hidden` with `body-scroll-lock` package. This package also deals with applying padding compensation for scrollbar of main page when needed. Since I am currently on a Mac can you please also verify that the scroll bar width is compensated for.
